higlass/trackdata.py

```diff
@@ -0,0 +1,76 @@
+import slugid
+from higlass.client import Track
+from pathlib import Path
+from typing import List, Tuple, Union
+from numpy import cumsum
+
+import itertools as it
+from pathlib import Path
+
+def chromsize_pairs(chromsizes_fn):
+    with open(chromsizes_fn, 'r') as f:
+        chroms = [l.strip().split('\t') for l in f.readlines()]
+        chromnames = [c[0] for c in chroms]
+        chromlengths = [int(c[1]) for c in chroms]
+    
+    return list(zip(chromnames, chromlengths))
+
+def load_chromsizes(chrom_names_lengths):
+    chromnames, chromlengths = zip(*chrom_names_lengths)
+
+    cumlengths = list(it.accumulate(chromlengths))
+    # we want the cumlengths to start 0
+    cumlengths = [0] + cumlengths[:-1]
+
+    return dict([(name, {
+        'name': name,
+        'length': length,
+        'start': start
+    }) for name, length, start in zip(chromnames, chromlengths, cumlengths)])
+
+def bedtiles(lines: List[Tuple[str, int, int, str, str, str]],
+    chroms: Union[Path, str, list], **kwargs) -> Track:
+    """Generate a list of local tiles that can be used with a bedlike track.
+    
+    Args:
+        lines: A list of bed style lines ([chrom, start, end, name, score, pos])
+        chromsizes: Either a Path or str pointing to a chromsizes file or a list
+            of [name, length] pairs.
+        
+    Returns:
+        A higlass Track that can be used with the viewer.
+    
+    """
+    if isinstance(chroms, Path) or isinstance(chroms, str):
+        chrom_names_lengths = chromsize_pairs(chroms)
+    if isinstance(chroms, list):
+        chrom_names_lengths = chroms
+    else:
+        ValueError(f"Unknown chroms type: {type(chroms)}. Expecting str, Path or list")
+
+    chroms = load_chromsizes(chrom_names_lengths)
```

Comment:
Why force the user to pass in a file?

Reply:
Oh, I see that it could be a list of pairs. I don't see why it can't just be a dictionary (pandas Series are also dict-like and dicts are now always ordered as of py3.6).

Reply:
It sure can. I added that as a supported option.